packages/project/src/db/project.ts

```diff
@@ -204,25 +205,40 @@ const clone = async (
       ? await db.build.loadByProjectId(project.id, "dev")
       : await db.build.loadByProjectId(project.id, "prod");
 
-  const projectId = uuid();
-  authorizeProject.registerProjectOwner({ projectId }, context);
+  const newProjectId = uuid();
+  await authorizeProject.registerProjectOwner(
+    { projectId: newProjectId },
+    context
+  );
 
   const clonedProject = await prisma.$transaction(async (client) => {
     const clonedProject = await client.project.create({
       data: {
-        id: projectId,
+        id: newProjectId,
         userId: userId,
         title: title ?? project.title,
         domain: generateDomain(project.title),
       },
     });
 
     await db.build.create(
-      { projectId: clonedProject.id, env: "dev", sourceBuild: build },
+      { projectId: newProjectId, env: "dev", sourceBuild: build },
       context,
       client
     );
 
+    await cloneAssets(
+      {
+        fromProjectId: project.id,
+        toProjectId: newProjectId,
+
+        // Permission check on newProjectId will fail until this transaction is committed.
```

Comment:
Will fail in the open-source version because in open source we take the owner from the Project table, and permission check is executed in other transaction.

Reply:
Not sure I understand. 

`dontCheckEditPermission: true` is here specifically to workaround this issue

Reply:
Or do you mean that the code comment should say that we have this issue only in open source version?